### PR TITLE
[DNM] Update matter bridge to NCS 2.4.0

### PR DESCRIPTION
**Pull request by markus-becker-tridonic-com**

### CODE REVIEW

**Comment by markus-becker-tridonic-com** [comment]
The signature change to the weak functions emberAfExternalAttributeReadCallback and emberAfExternalAttributeWriteCallback is not yet included, thus always the weak functions are called.

File: boards/nrf52840dk_nrf52840.overlay [hunk 1]
```diff
@@ -3,6 +3,13 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+/ {
+	chosen {
```
Comment (kkasperczyk-no) [inline boards/nrf52840dk_nrf52840.overlay:7]: If the sample is going to support OTA DFU on the nRF5340 DK, it needs the below code in its overlay as well.

File: src/app_task.cpp [hunk 2]
```diff
@@ -124,33 +146,43 @@ k_timer sFunctionTimer;
...
+                // Handle wrap condition
+                if (++gCurrentEndpointId < gFirstDynamicEndpointId)
+                {
+                    gCurrentEndpointId = gFirstDynamicEndpointId;
+                }
+            }
+        }
+        index++;
+    }
+    ChipLogProgress(DeviceLayer, "Failed to add dynamic endpoint: No endpoints available!");
```
Comment (kkasperczyk-no) [inline src/app_task.cpp:184]:
```suggestion
    ChipLogError(DeviceLayer, "Failed to add dynamic endpoint: No endpoints available!");
```

File: prj.conf [hunk 3]
```diff
@@ -4,104 +4,153 @@
...
 
+# This sample uses Kconfig.defaults to set options common for all
+# samples. This file should contain only options specific for this sample
+# or overrides of default values.
+
 # Enable CHIP
 CONFIG_CHIP=y
 CONFIG_CHIP_ENABLE_PAIRING_AUTOSTART=y
 CONFIG_CHIP_PROJECT_CONFIG="src/chip_project_config.h"
+# 32774 == 0x8006 (example lock-app)
```
Comment (kkasperczyk-no) [inline prj.conf:15]: How about 0x8000 = 32768 ? It is not allocated to any device type https://github.com/project-chip/connectedhomeip/blob/master/docs/examples/discussion/PID_allocation_for_example_apps.md and it has corresponding DAC in the examples.

File: prj.conf [hunk 4]
```diff
@@ -4,104 +4,153 @@
...
+CONFIG_BT_DEVICE_NAME="MatterBridge"
+
+# Stack size settings
+CONFIG_IEEE802154_NRF5_RX_STACK_SIZE=1024
+
+# Other settings
+CONFIG_THREAD_NAME=y
+CONFIG_MPU_STACK_GUARD=y
+CONFIG_RESET_ON_FATAL_ERROR=n
 # Enable Thread 1.2 features
```
Comment (kkasperczyk-no) [inline prj.conf:44]: Those configs are not necessary, as they are enabled by default.

File: prj.conf [hunk 5]
```diff
@@ -4,104 +4,153 @@
...
+CONFIG_OPENTHREAD_NORDIC_LIBRARY_MTD=y
 CONFIG_OPENTHREAD_MTD=y
 CONFIG_OPENTHREAD_FTD=n
-CONFIG_OPENTHREAD_SLAAC=y
-CONFIG_OPENTHREAD_MTD_NETDIAG=y
-CONFIG_OPENTHREAD_MANUAL_START=y
-CONFIG_OPENTHREAD_THREAD_STACK_SIZE=6144
-CONFIG_OPENTHREAD_NUM_MESSAGE_BUFFERS=64
 
+# Default OpenThread network settings
```
Comment (kkasperczyk-no) [inline prj.conf:28]: I think this configuration is not required. It used to be supported to form the default Thread network without commissioning, but the support was dropped.

File: src/app_task.cpp [hunk 6]
```diff
@@ -620,7 +658,9 @@ void AppTask::ZigbeeEventHandler(ZigbeeShell * shell, ZigbeeShell::Event_t event
 		for (auto &light : Lights)
 		{
 			if (!strcmp(light.GetName(), "none")) {
-				AddDeviceEndpoint(&light, &bridgedLightEndpoint, DEVICE_TYPE_LO_ON_OFF_LIGHT);
+				AddDeviceEndpoint(&light, &bridgedLightEndpoint,
```
Comment (kkasperczyk-no) [inline src/app_task.cpp:661]: Considering the fact this method may fail and return -1, I think the returned value should be somehow handled.

File: child_image/multiprotocol_rpmsg/prj_no_dfu.conf [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
```
Comment (kkasperczyk-no) [inline child_image/multiprotocol_rpmsg/prj_no_dfu.conf:2]: prj_no_dfu.conf and prj_release.conf files are not needed in the mcuboot and multiprotocol_rpmsg directories, as long, as we have only one prj.conf file in the main sample directory.

**Review by kkasperczyk-no — COMMENTED** [review]
Thank you a lot for the contribution! I have several initial comments.

**Comment by lats1980** [comment]
I got lots of compile error like:

C:\Users\lats\ncs\v2.0.2\modules\lib\matter\src\app\OperationalDeviceProxy.h:200:51: error: 'const struct chip::Dnssd::CommonResolutionData' has no member named 'interfaceId'
  200 |             interfaceId = nodeData.resolutionData.interfaceId;
      |                                                   ^~~~~~~~~~~
ninja: build stopped: subcommand failed.

Is there some missing code not included in the PR?

**Comment by markus-becker-tridonic-com** [comment]
> I got lots of compile error like:
> 
> C:\Users\lats\ncs\v2.0.2\modules\lib\matter\src\app\OperationalDeviceProxy.h:200:51: error: 'const struct chip::Dnssd::CommonResolutionData' has no member named 'interfaceId' 200 | interfaceId = nodeData.resolutionData.interfaceId; | ^~~~~~~~~~~ ninja: build stopped: subcommand failed.
> 
> Is there some missing code not included in the PR?

Yes, indeed. E.g. see https://github.com/NordicPlayground/matter-light-bridge/pull/1#issuecomment-1380528823 and I did some more changes (regen zap). Did not have the time yet to update the PR.

Since I do not have ZigBee devices I modified this sample app and added a shell for adding new devices. This makes it harder now to add the relevant changes back into this PR.

File: boards/nrf52840dk_nrf52840.overlay (hunk 1 above)
Comment (markus-becker-tridonic-com) [reply boards/nrf52840dk_nrf52840.overlay:7]: I have not looked at OTA at all.

File: src/app_task.cpp (hunk 2 above)
Comment (markus-becker-tridonic-com) [reply src/app_task.cpp:184]: Applied

File: prj.conf (hunk 4 above)
Comment (markus-becker-tridonic-com) [reply prj.conf:44]: removed.

File: src/app_task.cpp (hunk 6 above)
Comment (markus-becker-tridonic-com) [reply src/app_task.cpp:661]: done.

**Comment by markus-becker-tridonic-com** [comment]
> I got lots of compile error like:
> 
> C:\Users\lats\ncs\v2.0.2\modules\lib\matter\src\app\OperationalDeviceProxy.h:200:51: error: 'const struct chip::Dnssd::CommonResolutionData' has no member named 'interfaceId' 200 | interfaceId = nodeData.resolutionData.interfaceId; | ^~~~~~~~~~~ ninja: build stopped: subcommand failed.
> 
> Is there some missing code not included in the PR?

I have updated the PR to NCS 2.4.0. Would you be able to check?

File: prj.conf (hunk 3 above)
Comment (markus-becker-tridonic-com) [reply prj.conf:15]: done

File: prj.conf (hunk 5 above)
Comment (markus-becker-tridonic-com) [reply prj.conf:28]: removed.

**Comment by markus-becker-tridonic-com** [comment]
As a side-effect this might have changed the pins to the default pins for UART1 (1.00 and 1.01).

**Comment by markus-becker-tridonic-com** [comment]
The changes were done in https://github.com/NordicPlayground/matter-light-bridge/pull/2.

**Pull request closed without merging**